src/algebra/lie/basic.lean

```diff
@@ -562,6 +562,9 @@ def lie_module.to_endo_morphism : L →ₗ⁅R⁆ module.End R M :=
   map_smul' := λ t x, by { ext m, apply smul_lie, },
   map_lie   := λ x y, by { ext m, apply lie_lie, }, }
 
+@[simp] lemma lie_module.to_endo_morphism_apply (x : L) (m : M) :
```

Comment:
I think you can add `@[simps]` on the def above to get this automatically - but maybe it picks the wrong name.

Reply:
Thanks; I had forgotten about `simps`! Trying now, I see it names the lemma `lie_module.to_endo_morphism_to_fun_apply` which is fine by me.

Reply:
You can train simps to use `apply` instead of `to_fun` there - have a look at how we do it for equiv and linear maps.

Reply:
Oh I see we even have a [library note](https://leanprover-community.github.io/mathlib_docs/notes.html#custom%20simps%20projection) on this.

I just tried:
```lean
initialize_simps_projections lie_module.to_endo_morphism (to_fun_apply → apply)
```
and it just says `failed`.

Is it really worth going to trouble to persuade `simps` to name things differently in this case? I'm happy with `_to_fun_apply` and if I really want to control the name, I'd rather just have a one-liner defining a `rfl` lemma than grappling with meta code.

Reply:
It should be something more like
```
initialize_simps_projections lie_algebra.morphism (to_fun → apply)
```

Reply:
Thanks, I finally get the idea. Now that I understand this better, I would like to use `initialize_simps_projections` except that it slightly changes the type of lemma that is generated by having different coercions in place❗️

Specifically, without `initialize_simps_projections`, I get a lemma named `lie_module.to_endo_morphism_to_fun_apply` which states:
```lean
⇑(⇑(lie_module.to_endo_morphism R L M) x) m = ⁅x,m⁆
```
However, with `initialize_simps_projections`, I get a lemma named `lie_module.to_endo_morphism_apply_apply` which states:
```lean
⇑((lie_module.to_endo_morphism R L M).to_fun x) m = ⁅x,m⁆
```

The first lemma (with the coercion instead of `to_fun`) is preferable; indeed in the proof of `lie_module.mem_ker` I have to change `simp` to `simpa` to close the goal if I use `lie_module.to_endo_morphism_apply_apply`.

Reply:
That shouldn't be happening - I think you can fix it by using `initialize_simps_projections` _after_ creating the `has_coe_to_fun` instance.

Reply:
I can confirm that invoking `initialize_simps_projections` after creating the `has_coe_to_fun` resolves this issue, so I've pushed a commit which does this now. This is exactly the sort of meta code issue that makes me wary but perhaps I'm just too paranoid.

Reply:
I agree it's a little fragile, but we already use `initialize_simps_projections` on all the other bundled homs, so it's worth putting up with the quirks for consistency with `add_monoid_hom`, `linear_map`, `equiv`, ...

Reply:
Definitely in favour of consistency!